Content.Server/Ghost/Components/GhostComponent.cs

```diff
@@ -38,6 +43,15 @@ public bool CanReturnToBody
                 Dirty();
             }
         }
+        public bool CanGhostInteract
+        {
+            get => _canGhostInteract;
+            set
+            {
+                _canGhostInteract = value;
+                Dirty();
```

Comment:
You're calling Dirty here but you're not adding this variable to the ghost state.
Our system sadly cannot rely on magic, we need to be explicit about the variables we sync in states

Reply:
Should also guard it with if (_canGhostInteract == value) return;

Reply:
Also SharedGhostComponent has this property; you should override it instead.